pandas/tests/indexes/test_multi.py

```diff
@@ -204,6 +204,25 @@ def assert_matching(actual, expected):
         assert_matching(ind2.levels, new_levels)
         assert_matching(self.index.levels, levels)
 
+        # illegal level changing should not change levels
+        # GH 13754
+        original_index = self.index.copy()
+        with assertRaisesRegexp(ValueError, "^On"):
+            self.index.set_levels(['c'], level=0, inplace=True)
+        assert_matching(self.index.levels, original_index.levels)
+
+        with assertRaisesRegexp(ValueError, "^On"):
+            self.index.set_labels([0, 1, 2, 3, 4, 5], level=0, inplace=True)
+        assert_matching(self.index.labels, original_index.labels)
+
+        with assertRaisesRegexp(TypeError, "^Levels"):
+            self.index.set_levels('c', level=0, inplace=True)
+        assert_matching(self.index.levels, original_index.levels)
```

Comment:
can you add tests for ``inplace=False`` as well (you can loop over the param I think)